dashboard/src/components/AppList/AppList.tsx

```diff
@@ -30,19 +33,38 @@ class AppList extends React.Component<IAppListProps, IAppListState> {
     this.setState({ filter });
   }
 
-  public componentWillReceiveProps(nextProps: IAppListProps) {
+  public componentDidUpdate(prevProps: IAppListProps) {
     const {
       apps: { error, listingAll },
       fetchApps,
       filter,
       namespace,
     } = this.props;
     // refetch if new namespace or error removed due to location change
-    if (nextProps.namespace !== namespace || (error && !nextProps.apps.error)) {
-      fetchApps(nextProps.namespace, listingAll);
+    if (prevProps.namespace !== namespace || (error && !prevProps.apps.error)) {
+      fetchApps(namespace, listingAll);
     }
-    if (nextProps.filter !== filter) {
-      this.setState({ filter: nextProps.filter });
+    if (prevProps.filter !== filter) {
```

Comment:
We might actually want to refactor this to use the `defaultFilter` approach described in https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#alternative-2-reset-uncontrolled-component-with-an-instance-method, or try to make it fully-controlled.

It's things like this that worry me about changing this to `componentDidUpdate` in this PR, as we haven't fully thought-out what we should replace `componentWillReceiveProps` with and it's not relevant to this PR. I would consider reverting this to continue using `componentWillRecieveProps` and leave a comment to look into this later. It's not like we can't use `componentWillRecieveProps` so switching it to `componentDidUpdate` doesn't really give us anything, we still need to come back here at some point and figure out how to fix this.

Reply:
@andresmgot yes, we need to remove it when we update React, however we're still currently using React 16, so we can still use this. My main concern is not falling into the trap of thinking this is fixed when we should actually go back and refactor this. That said, it's not a huge issue to change this to `componentDidUpdate` as long as we add a TODO here.